### PR TITLE
CB-11244: Added logic for Android Studio Projects

### PR DESCRIPTION
**Pull request by infil00p**
This still needs tests, but this PR will allow for plugins to be installed on a Android Studio Project structure.  I was going to add tests, but ConfigFile currently has no tests, and it'd be good if I could get help on this.

### CODE REVIEW

File: src/ConfigChanges/ConfigFile.js [hunk 1]
```diff
@@ -184,29 +184,36 @@ function resolveConfigFilePath (project_dir, platform, file) {
...
 
-    // special-case config.xml target that is just "config.xml". This should be resolved to the real location of the file.
-    // TODO: move the logic that contains the locations of config.xml from cordova CLI into plugman.
+    // XXX this checks for android studio projects
+    // only if none of the options above are satisfied does this get called
+    // TODO: Move this out of cordova-common and into the platforms somehow
+    if (platform === 'android' && !fs.existsSync(filepath)) {
+        if (file === 'AndroidManifest.xml') {
+            filepath = path.join(project_dir, 'app', 'src', 'main', 'AndroidManifest.xml');
+        } else if (file.endsWith('config.xml')) {
```
Comment (stevengill) [inline src/ConfigChanges/ConfigFile.js]: nice use of `endsWith`. Looks like node added support for it in 4.x. So we are all good to use it! http://node.green/#ES2015-built-in-extensions-String-prototype-methods-String-prototype-endsWith

**Comment by infil00p** [comment]
@stevengill Why does this break on an old Android project? Basically, this will always go to wherever the config file is first, which is the old project structure.  If the file is found, it will edit that config file, otherwise it will go to the Android Studio location of that file.  We're not breaking backwards compatibility with this change AFAIK.

**Comment by audreyso** [comment]
Great! I will put cordova-common tests on my to-do list.

**Comment by audreyso** [comment]
Will these be unit tests or integration tests? @stevengill @infil00p

**Comment by infil00p** [comment]
@audreyso These will be unit tests, I think.

**Comment by stevengill** [comment]
Yup, unit

**Comment by audreyso** [comment]
@infil00p @stevengill Hi! I created unit tests for ConfigFile --> https://github.com/apache/cordova-common/pull/8. Let me know if this is okay! Thank you!

**Comment by stevengill** [comment]
Code looks good but some eslint failures going on

**Pull request closed without merging**